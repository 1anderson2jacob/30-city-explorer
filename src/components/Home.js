import React, { Component } from 'react';


class Home extends Component {

  state = {
    userLocation: ''
  }

  handleSubmit = e => {
    e.preventDefault();
    this.props.history.push(`/${this.state.userLocation}`);
  }
  handleChange = e => {
    this.setState({ userLocation: e.target.value })
  }


  render() {
    return (
      <div>
        <h1>City Exporer</h1>
        <h4>Enter a Location</h4>
        <form onSubmit={this.handleSubmit}>
          <input type="text" onChange={this.handleChange} placeholder="enter a location" />
          <button>Explore!</button>
        </form>
      </div>
    );
  }
}

export default Home;