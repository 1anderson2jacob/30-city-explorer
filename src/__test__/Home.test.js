import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });
import Home from '../components/Home';

describe('running tests on city explore Home', () => {
  it('is alive', () => {
    let component = shallow(<Home />);
    expect(component.find('h1').exists()).toBeTruthy();
  });
  it('renders correctly', () => {
    let render = renderer.create('<Home/>').toJSON();
    expect(render).toMatchSnapshot();
  });

});
