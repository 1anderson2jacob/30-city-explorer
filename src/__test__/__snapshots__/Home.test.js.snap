// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`running tests on city explore Home renders correctly 1`] = `"<Home/>"`;
